src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt

```diff
@@ -194,70 +198,77 @@ class ZaakafhandelParameterBeheerService @Inject constructor(
             referentieTabellen = it.referentieTabellen
             formulierDefinitieID = it.formulierDefinitieID
         }
-    }.toSet().let(nieuweZaakafhandelParameters::setHumanTaskParametersCollection)
+    }.toSet().let(newZaakafhandelParameters::setHumanTaskParametersCollection)
 
     /**
      * Kopieren van de UserEventListenerParameters van de oude ZaakafhandelParameters naar de nieuw ZaakafhandelParameters
      *
-     * @param vorigeZaakafhandelparameters bron
-     * @param nieuweZaakafhandelParameters bestemming
+     * @param previousZaakafhandelParameters bron
+     * @param newZaakafhandelParameters bestemming
      */
     private fun mapUserEventListenerParameters(
-        vorigeZaakafhandelparameters: ZaakafhandelParameters,
-        nieuweZaakafhandelParameters: ZaakafhandelParameters
-    ) = vorigeZaakafhandelparameters.userEventListenerParametersCollection.map {
+        previousZaakafhandelParameters: ZaakafhandelParameters,
+        newZaakafhandelParameters: ZaakafhandelParameters
+    ) = previousZaakafhandelParameters.userEventListenerParametersCollection.map {
         UserEventListenerParameters().apply {
             planItemDefinitionID = it.planItemDefinitionID
             toelichting = it.toelichting
         }
-    }.toSet().let(nieuweZaakafhandelParameters::setUserEventListenerParametersCollection)
+    }.toSet().let(newZaakafhandelParameters::setUserEventListenerParametersCollection)
 
     /**
      * Kopieren van de ZaakbeeindigGegevens van de oude ZaakafhandelParameters naar de nieuw ZaakafhandelParameters
      *
-     * @param vorigeZaakafhandelparameters bron
-     * @param nieuweZaakafhandelParameters bestemming
-     * @param nieuwZaaktype                het nieuwe zaaktype om de resultaten van te lezen
+     * @param previousZaakafhandelParameters bron
+     * @param newZaakafhandelParameters bestemming
+     * @param newZaaktype                het nieuwe zaaktype om de resultaten van te lezen
      */
     private fun mapZaakbeeindigGegevens(
-        vorigeZaakafhandelparameters: ZaakafhandelParameters,
-        nieuweZaakafhandelParameters: ZaakafhandelParameters,
-        nieuwZaaktype: ZaakType
+        previousZaakafhandelParameters: ZaakafhandelParameters,
```

Comment:
This one could be rewritten as a re-mapping into the same `ZaakafhandelParameters`

```kotlin
    private fun remapZaakbeeindigGegevens(
        zaakafhandelParameters: ZaakafhandelParameters,
        newZaaktype: ZaakType
    ) {
        val newResultaattypen = newZaaktype.resultaattypen.map { ztcClientService.readResultaattype(it) }
        zaakafhandelParameters.nietOntvankelijkResultaattype =
            zaakafhandelParameters.nietOntvankelijkResultaattype?.let {
                mapVorigResultaattypeOpNieuwResultaattype(it, newResultaattypen)
            }
        val zaakbeeindigParametersCollection = zaakafhandelParameters.zaakbeeindigParameters.mapNotNull {
                zaakbeeindigParameter ->
            zaakbeeindigParameter.resultaattype
                ?.let { mapVorigResultaattypeOpNieuwResultaattype(it, newResultaattypen) }
                ?.let {
                    ZaakbeeindigParameter().apply {
                        zaakbeeindigReden = zaakbeeindigParameter.zaakbeeindigReden
                        resultaattype = it
                    }
                }
        }.toMutableSet()
        zaakafhandelParameters.setZaakbeeindigParameters(zaakbeeindigParametersCollection)
    }

```

Reply:
Or maybe as an extension function on `ZaakafhandelParameters` itself and only pass the new case type...